base/initdefs.jl

```diff
@@ -136,10 +136,10 @@ function init_load_path()
     end
     project = (JLOptions().project != C_NULL ?
         unsafe_string(Base.JLOptions().project) :
-        get(ENV, "JULIA_PROJECT", nothing))
+        get(ENV, "JULIA_PROJECT", ""))
```

Comment:
This has a behavioral side effect: it makes setting `JULIA_PROJECT` to the empty string behave the same as having it unset, which doesn't seem ideal and is something that I've generally avoided.